director/redirect.go

```diff
@@ -402,8 +402,13 @@ func registerServeAd(ctx *gin.Context, sType ServerType) {
 		prefix := path.Join("caches", ad.Name)
 		ok, err := VerifyAdvertiseToken(token, prefix)
 		if err != nil {
-			log.Warningln("Failed to verify token:", err)
-			ctx.JSON(400, gin.H{"error": "Authorization token verification failed"})
+			if !ok {
+				log.Warningln("Failed to verify token:", err)
+				ctx.JSON(404, gin.H{"error": "Cache is not admin approved"})
+			} else {
+				log.Warningln("Failed to verify token:", err)
+				ctx.JSON(400, gin.H{"error": "Authorization token verification failed"})
```

Comment:
I don't think we need to change anything here, but we should make sure that the `400` (`http.StatusBadRequest`) response is only reachable in the event that the error is due to the advertise token itself being malformed. From the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400), `400` is for bad client requests.